packages/framework-provider-local-infrastructure/test/scheduler.test.ts

```diff
@@ -0,0 +1,145 @@
+import { BoosterConfig } from '@boostercloud/framework-types'
+import { describe } from 'mocha'
+import { fake, replace, restore } from 'sinon'
+import { configureScheduler } from '../src/scheduler'
+import { expect } from '../test/expect'
+import * as scheduler from 'node-schedule'
+import * as fc from 'fast-check'
+import { isValidCron } from 'cron-validator'
+
+describe('scheduler', () => {
+  let mockConfig: any
+
+  const userProject = {
+    boosterTriggerScheduledCommand: () => {},
+  }
+
+  describe('configureScheduler', () => {
+    beforeEach(() => {
+      replace(scheduler, 'scheduleJob', fake())
+    })
+
+    afterEach(() => {
+      restore()
+    })
+
+    it('should only schedule commands where `scheduledOn` exists and is not undefined or null or empty', () => {
+      mockConfig = {
+        scheduledCommandHandlers: {
+          cmd1: {
+            class: fake(),
+            scheduledOn: {
+              minute: '5',
+              hour: '4',
+              day: '*',
+              month: '*',
+              weekDay: '*',
+              year: '*',
+            },
+          },
+          cmd2: {
+            class: fake(),
+            scheduledOn: {
+              minute: '*',
+              hour: '4',
+              day: '3',
+              month: '*',
+              weekDay: '*'
+            },
+          },
+          cmdWithoutScheduledOn: {
+            class: fake(),
+          },
+          cmdWithEmptyScheduledOn: {
+            class: fake(),
+            scheduledOn: {},
+          },
+          cmdWithUndefinedScheduledOn: {
+            class: fake(),
+            scheduledOn: undefined,
+          },
+          cmdWithNullScheduledOn: {
+            class: fake(),
+            scheduledOn: null,
+          },
+        },
+      }
+
+      configureScheduler(mockConfig as unknown as BoosterConfig, userProject)
+
+      expect(scheduler.scheduleJob).to.have.been.calledWithMatch('cmd1', '5 4 * * *')
+      expect(scheduler.scheduleJob).to.have.been.calledWithMatch('cmd2', '* 4 3 * *')
+      // expect(scheduler.scheduleJob).to.have.been.calledWithMatch('cmdWithEmptyScheduledOn', {})
```

Comment:
This assertion was passing before the change here:
https://github.com/boostercloud/booster/pull/1169/files#diff-0871e6dff27016ddf8f5c7753a016ebcf1717fa0bfcd7bad37bd6c06de4029b3R18

If that change is correct, then this commented out assertion should be removed; if the change is wrong, this assertion should be re-enabled and its opposite assertion (`expect(scheduler.scheduleJob).to.not.have.been.calledWithMatch('cmdWithEmptyScheduledOn', {})`) should be removed instead.